lib/reek/context/code_context.rb

```diff
@@ -95,16 +98,12 @@ def each(&block)
 
       alias_method :parent, :context
 
-      # Register a child context. The child's parent context should be equal to
-      # the current context.
-      #
-      # This makes the current context responsible for setting the child's
-      # visibility.
+      # Register a child context. Used by a child context to register itself with its parent.
```

Comment:
I'll update this to explain it better. In short, the child is supposed to call this method on its parent.